### PR TITLE
⬆️ upgrading docker-compse to 2.x.x in dynamic-sidecar 🚨

### PR DESCRIPTION
**Pull request by GitHK**
## What do these changes do?

- ♻️ `docker-compose` was replaced with `docker compose` inside the `dynamic-sidecar`
- ♻️ `compose-spec` upgraded network format which was deprecated. `external` is no longer a dic.
- ♻️ `docker-compose` python package is no longer a dependency of the `dynamic-sidecar`, it is now installed via apt.
- ♻️ unpinned all constraints form the `dynamic-sidecar`'s dependencies.

Bonus:
- 🎯 pinned `docker-ce-cli` and `docker-compose-plugin` versions in the docker images of the `autoscaling` and `dynamic-sidecar` services.
- ⬆️ upgraded `PyYAML==6.0.1` in the `dynamic-sidecar`

Checks:

- [ ] 🚨 check in master that all multi container services start and stop as expected

## Related issue/s

- https://github.com/ITISFoundation/osparc-issues/issues/638
- https://github.com/ITISFoundation/osparc-simcore/pull/4503
- fixes https://github.com/ITISFoundation/osparc-simcore/issues/4507

### CODE REVIEW

**Review by matusdrobuliak66 — APPROVED** [review]
:+1: please also add 🚨 to the name of the PR, so we know we should be careful. Thanks!

File: services/autoscaling/Dockerfile [hunk 1]
```diff
@@ -32,11 +33,10 @@ RUN --mount=type=cache,target=/var/cache/apt,mode=0755,sharing=private \
   && apt-get update \
   && apt-get install -y --no-install-recommends \
   # only the cli is needed and we remove the unnecessary stuff again
-  docker-ce-cli \
+  docker-ce-cli=${DOCKER_APT_VERSION} \
   && apt-get remove -y\
   ca-certificates \
   curl \
-  gnupg \
   lsb-release \
```
Comment (sanderegg) [inline services/autoscaling/Dockerfile:40]: why this change?

**Review by sanderegg — APPROVED** [review]
Thanks! please also check if cython really needs to be installed with @pcrespov. I think there is no need to install unneeded packages.

File: services/autoscaling/Dockerfile (hunk 1 above)
Comment (GitHK) [reply services/autoscaling/Dockerfile:40]: it's already installed above, no need to reinstall it

File: services/autoscaling/Dockerfile [hunk 2]
```diff
@@ -12,6 +12,7 @@ FROM python:${PYTHON_VERSION}-slim-buster as base
 
 LABEL maintainer=sanderegg
 
+ENV DOCKER_APT_VERSION="5:24.0.5-1~debian.10~buster"
```
[the remark this answers is not in the document]
Comment (GitHK) [reply services/autoscaling/Dockerfile:15]: This is 1 specific version. I wanted it to be fixed.
To upgrade is pretty simple. `apt-cache madison docker-ce | awk '{ print $3 }'` will give you the latest versions. I guess we can have some sort of script for listing them.

File: services/dynamic-sidecar/requirements/_base.in [hunk 3]
```diff
@@ -30,7 +30,6 @@ aiofiles
 aioprocessing
 arrow
 cython # Explicitly added to build pyyaml (see constraints)
```
[the remark this answers is not in the document]
Comment (GitHK) [reply services/dynamic-sidecar/requirements/_base.in:32]: yes this one can be dropped as well

File: services/autoscaling/Dockerfile (hunk 2 above)
[the remark this answers is not in the document]
Comment (GitHK) [reply services/autoscaling/Dockerfile:16]: @sanderegg I've added some scripts to aid with the upgrade and some notes on both images.

File: services/autoscaling/Dockerfile (hunk 1 above)
Comment (sanderegg) [reply services/autoscaling/Dockerfile:42]: this is about removing it
